### DISCUSSION TITLE
Buffer protocol for jaxlib.xla_extension.Array

### DISCUSSION THREAD

**Discussion by heiner** [jax-ml/jax#14712] [General] [answered]
Googling [suggests](https://stackoverflow.com/questions/30017991/check-if-an-object-supports-the-buffer-protocol-python) a Python version of `PyObject_CheckBuffer` could look like

```python
def check_buffer_doesntwork(obj):
    try:
        memoryview(obj)
    except TypeError:
        return False
    return True
```

This works for `np.ndarray`s and various other types, but not for `jaxlib.xla_extension.Array`:

```
>>> memoryview(jnp.arange(10))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: memoryview: a bytes-like object is required, not 'Array'
```

Is this intentional?

**Comment by hawkinsp** [comment, marked as answer]
Hmm. We used to support this. I think it might have gotten lost as part of the unified jax.Array transition in 0.4. Let me look into it.

**Reply by hawkinsp** [reply]
Created #14713 to track.

**Reply by heiner** [reply]
Thanks! Great turnaround time! :)
